genericjpa/db/src/main/java/org/hibernate/search/annotations/UpdateInfo.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.annotations;
+
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * @author Martin Braun
+ * @hsearch.experimental
+ */
+@Target({FIELD, METHOD, TYPE})
+@Retention(RUNTIME)
+@Repeatable(UpdateInfos.class)
+public @interface UpdateInfo {
```

Comment:
Additionally the JPA metamodel only gives information about the Java side of things. No notion of where the information is stored on the database level.

With using the annotations as a fallback (NOT with Hibernate ORM, but as I said for the sake of this PR I want to leave it as is and enlist the work of someone that knows the Hibernate Persister better than me for a follow up issue) we make sure every JPA provider is supported. Even small ones like BatooJPA (which I do not intend writing a custom integration for). For the big JPA implementations I intend to have a custom solution that works with the least amount of extra work possible.